code/modules/projectiles/guns/mods/mods.dm

```diff
@@ -12,6 +12,38 @@
 /obj/item/gun_upgrade/muzzle
 	bad_type = /obj/item/gun_upgrade/muzzle
 
+/obj/item/gun_upgrade/underbarrel
+	bad_type = /obj/item/gun_upgrade/underbarrel
+
+/obj/item/gun_upgrade/underbarrel/foregrip
+	name = "foregrip"
+	desc = "A fancy rubber grip that reduces recoil during firing when installed under the gun barrel. However, it also makes bracing impossible."
+	icon_state = "foregrip"
+	rarity_value = 15
+
+/obj/item/gun_upgrade/underbarrel/foregrip/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_FOREGRIP = TRUE,
+		GUN_UPGRADE_RECOIL = 0.8
+		)
+	I.gun_loc_tag = GUN_UNDERBARREL
+
+/obj/item/gun_upgrade/underbarrel/bipod
+	name = "bipod"
+	desc = "A simple set of telescopic poles to keep a weapon stabilized during firing. It greatly reduces recoil when deployed, but also increases the gun\'s weight, making it heavy to carry."
+	icon_state = "bipod"
+	rarity_value = 15
+
+/obj/item/gun_upgrade/underbarrel/bipod/New()
+	..()
+	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
+	I.weapon_upgrades = list(
+		GUN_UPGRADE_BIPOD = TRUE
+		)
```

Comment:
According to design doc, bipod should increase recoil when held in hand instead of cause a slowdown.
```suggestion
		GUN_UPGRADE_RECOIL = 1.2
		)
```